src/shared-validator.h

```diff
@@ -124,6 +124,7 @@ class SharedValidator {
   Result EndBrTable(const Location&);
   Result OnCall(const Location&, Var func_var);
   Result OnCallIndirect(const Location&, Var sig_var, Var table_var);
+  Result OnCallRef(const Location&, Index* function_type_index = nullptr);
```

Comment:
Why does the second arg need to have a default here?  I only see a single call site and it passes a non-null arg here?    Could we remove this and the check in OnCallRef?   In fact could we pass by ref?

Reply:
Removed default value, but preserved pointer because it is little bit readable at the call site